runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/OpAdapter.java

```diff
@@ -102,7 +115,16 @@ public synchronized CompletionStage<Collection<OpMessage<OutT>>> apply(OpMessage
               String.format("Unexpected input type: %s", message.getType()));
       }
     } catch (Exception e) {
-      LOG.error("Op {} threw an exception during processing", this.getClass().getName(), e);
+      LOG.error("Op {} threw an exception during processing", transformFullName, e);
+      e.addSuppressed(
+          new RuntimeException(
+              String.format("Op: %s threw an exception during processing", transformFullName)));
+      exceptionListeners.forEach(
```

Comment:
Please add another try-catch {ignore} here for exceptionListeners.forEach. I don't want this block to be interrupted by the listeners.

You might consider move this logic line 119 - 127 in a private function, e.g. notifyExceptionListners(...)

Reply:
good call, done